h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeSubgraph.java

```diff
@@ -59,7 +60,7 @@ public void setFontSize(int fontsize) {
    * @return The new child node
    */
   public SharedTreeNode makeLeftChildNode(SharedTreeNode parent) {
-    SharedTreeNode child = new SharedTreeNode(parent, subgraphNumber, parent.getDepth() + 1);
+    SharedTreeNode child = new SharedTreeNode(nodesArray.size(), parent, subgraphNumber, parent.getDepth() + 1);
```

Comment:
This does not make sense to me. Node's ID is length of the array with nodes ? Does the array grow, so the node ID always corresponds to the array size ?

Reply:
@Pscheidl the arraylist grows, adding a node to the list is thus equivalent to incrementing a node id